PurchasesCoreSwift/Purchasing/ReceiptFetcher.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  ReceiptFetcher.swift
+//  Purchases
+//
+//  Created by Javier de Martín Gil on 8/7/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+// TODO: Make internal after migration to Swift is complete
+@objc(RCReceiptFetcher) public class ReceiptFetcher: NSObject {
+    private let requestFetcher: StoreKitRequestFetcher
+    private let receiptBundle: Bundle
+
+    @objc public convenience init(requestFetcher: StoreKitRequestFetcher) {
+        self.init(requestFetcher: requestFetcher, bundle: .main)
+    }
+
+    init(requestFetcher: StoreKitRequestFetcher, bundle: Bundle) {
+        self.requestFetcher = requestFetcher
+        self.receiptBundle = bundle
+    }
+
+    @objc public func receiptData(refreshPolicy: ReceiptRefreshPolicy,
+                                  completion: @escaping ((Data?) -> Void)) {
+        if refreshPolicy == .always {
+            Logger.debug(String(format: Strings.receipt.force_refreshing_receipt))
+            self.refreshReceipt(completion)
```

Comment:
One of the things I've noticed that we do a lot is `self.` when `self` isn't required. I'd like to propose that we omit `self` for all things unless required in closures, or same-name params / instance vars or vars with the same name as funcs.

Reply:
I usually omit it but sometimes I accidentally leave it in when moving code away from a block for example. It'd be nice to get swiftlint to assist. 
It looks like [this should do the trick](https://realm.github.io/SwiftLint/explicit_self.html)

Reply:
looks like there's been a [request for it in swiftlint since 2015](https://github.com/realm/SwiftLint/issues/59) so I wouldn't hold my breath 😂
we can manually check for now. AppCode does offer it as a feature, though.